crates/pico_core/src/storage.rs

```diff
@@ -0,0 +1,25 @@
+use std::any::Any;
+
+use crate::{
+    container::Container,
+    database::Database,
+    epoch::Epoch,
+    key::Key,
+    node::{Dependency, DerivedNode, DerivedNodeId, SourceNode},
+    params::ParamId,
+};
+
+pub trait Storage<Db: Database + ?Sized> {
+    fn derived_nodes(&self) -> &impl Container<DerivedNodeId, DerivedNode<Db>>;
```

Comment:
might be worth commenting that it is safe to GC these *as long as we're not currently calling a memoized function*